deepchecks/vision/utils/image_properties.py

```diff
@@ -122,6 +132,25 @@ def get_dimension(img) -> int:
     return img.shape[2]
 
 
+def calc_default_image_properties(batch: List[np.ndarray]) -> Dict:
+    """Speed up the calculation for the default image properties by sharing common actions."""
+    results_dict = {}
+    sizes_array = _sizes_array(batch)
+    results_dict['Aspect Ratio'] = list(sizes_array[:, 0] / sizes_array[:, 1])
+    results_dict['Area'] = list(sizes_array[:, 0] * sizes_array[:, 1])
+
+    grayscale_images = [rgb2gray(img) if _is_grayscale(img) else img for img in batch]
+    results_dict['Brightness'] = [image.mean() for image in grayscale_images]
```

Comment:
Why not use the existing code for brightness etc? We're still iterating on the images here, so we're not saving any time here except with the _rgb_relative_intensity_mean_array

Reply:
We're saving time by calling _is_grayscale once for both brightness and RMS